### ISSUE TITLE
CLI on Windows: --folder-uri option only works as the last argument and when the --wait option is not present

### ISSUE THREAD

**Issue by kiwi-bui** [microsoft/vscode#209072]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.87.2
- OS Version: Windows 10 Pro 19044.2486 VM

Steps to Reproduce:
1.  Case where --folder-uri is not the last option. Run `code --folder-uri vscode-remote://ssh-remote+vscode@some-workspace --disable-extension ms-vscode-remote.remote-containers --new-window --wait`. Command will return immediately without action or error.
2. Case where --folder-uri is the last option but --wait is present. Run `code  --disable-extension ms-vscode-remote.remote-containers --new-window  --wait --folder-uri vscode-remote://ssh-remote+vscode@some-workspace`. Command will return immediately without action or error.

There seems to be a bug where the the code command silently errors when --folder-uri and the --wait options are used together. Also, there seems to be a bug where the the code command silently errors when the --folder-uri option is not the last option.

**Comment by connor4312** [comment]
This seems like an electron bug(?)

Passing `code --any-arg with-a-colon: --verbose` fails (doesn't call `main.js`) but removing the colon works `code --any-arg without-a-colon --verbose`. Same when calling the exe directly versus going through a batch script.

**Comment by deepak1556** [comment]
Yeah the uri argument handling is not a bug but a security fix in the runtime, we also hit this in another feature of vscode for context: https://github.com/microsoft/vscode/issues/188769#issuecomment-1727691561

Would it be possible to encode the uri for the argument ?

**Comment by kiwi-bui** [comment]
Is this related to the --wait option not being able to be used with the --folder-uri option?

**Comment by kazoogh** [comment]
Hello! I am looking for a good first issue to contribute to as practice and this seems like one I can work on. Will be working on this for now hope that is okay :)

**Comment by zhanglinqian** [comment]
I'd like to work on this issue.
